File: Mathematical Algorithms/Amicable Pair/Amicable.java
public class Amicable {
    static int divSum(int n) {

        int result = 0;

        for (int i = 2; i <= Math.sqrt(n); i++) {
            // if 'i' is divisor of 'n'
            if (n % i == 0) {
                // if both divisors are same
                // then add it once else add
                // both
                if (i == (n / i))
                    result += i;
                else
                    result += (i + n / i);
            }
        }

        // Add 1 and n to result as above loop
        // considers proper divisors greater
        // than 1.
        return (result + 1);
    }

    static boolean areAmicable(int x, int y) {
        if (divSum(x) != y)
            return false;

        return (divSum(y) == x);
    }
}

File: Mathematical Algorithms/Amicable Pair/Amicable.py
import math


def divSum(n):
    # Sum of divisors
    result = 0

    # find all divisors
    # which divides 'num'
    for i in range(2, int(math.sqrt(n)) + 1):

        # if 'i' is
        # divisor of 'n'
        if n % i == 0:

            # if both divisors are same
            # then add it once else add
            # both
            if i == int(n / i):
                result = result + i
            else:
                result = result + (i + int(n / i))

    return result + 1


def areAmicable(x, y):
    if divSum(x) != y:
        return False

    return divSum(y) == x